.github/workflows/compat.yml

```diff
@@ -49,7 +49,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ts: ['4.7', '4.8', '4.9', '5.0', '5.1', '5.2']
+        ts: ['4.7', '4.8', '4.9', '5.0', '5.1', '5.2', '5.3']
```

Comment:
Does anybody know if TypeScript versions have EOL by design? 

I really wish we dropped the left-most version when adding the latest version support. But on its own that's a breaking change, and I hate to spawn countless major versions of MSW (the TS team moves _really_ fast). If they have the concept of EOL, perhaps that can help us reason that dropping dead versions isn't technically a breaking change since nobody should be using those.

Reply:
They don't implement semver in that way and don't keep fixing older versions generally

I think we're probably ok to be additive or even just set 4.7 as min and keep moving.  

If we start shipping a type that doesn't work with 4.7 we can bump that and consider it a breaking change, but generally version to version breakage from typescript should be quite limited I would think?


Fwiw, I don't know of many libraries that test against that many versions or enforce it as exactly, so we could consider

Test minimum supposed
Test Max version
Assume everything between works
Save time and money on ci
If minimum breaks, bump the minimum and consider that a breaking change?

Reply:
Basically every release has a change that might break something, whether it's types consumer (lib dom, es..., etc) or some subtle compilation bug, so every version is somewhat breaking, but mostly those are consumer fixable, so we could probably just set a minimum occasionally too and deal with something that doesn't work in the future when it doesn't

It's much less common to have real incompatible changes, than things _just working_?

Reply:
Well, there are some projects that follow a support window, with the most important one being [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/#support-window), and another one could be [typescript-eslint](https://typescript-eslint.io/users/dependency-versions#typescript).

Reply:
> but generally version to version breakage from typescript should be quite limited I would think?

The practice has proved this not to be the case. I found issues between 4.5 and 4.6, I believe, and TS can and often does ship breaking changes between minor versions (you're right, they aren't semver). Because of that, we shouldn't be assuming everything from X to Y automatically works. 

> but mostly those are consumer fixable

Not always. MSW can declare types in a way that works on one version of TS while failing to compile the user's app on the other. It's an MSW bug, not the user's. 

It's sad if there are no official dead versions of TS though.

Reply:
I think, as a consumer, id probably prefer to be able to npm install and see failures than have npm install fail, so I can start making updates to other code, which our current style of over enforcing versions for typescript doesn't make ergonomic


I'd suggest opening up the peers, and handling errors when they happen vs assuming they happen

Reply:
sounds like a good decision document to me.

1. why we are opening the peer dependencies.
2. how to address ts-related issues (by adding the typing test target for the problematic ts version; versions without explicit tests are considered working).

this is more of a development style decision. if we allow anything and approach this on a failure-based basis, we are opening ourselves to the flow of ts issues (agree, _potential_) that people will be demanding to resolve. vs saying "no" to any issues on officially unsupported TS versions, giving us time to add automated tests for them before promising the support. what you propose is an optimistic promise. what we have now is overly-precautious denial of promise. but if we have active contributors such as yourself, i see little problem in switching the approach to what you suggest.

Reply:
Agreed, should we try to land thisto unblock consumers from upgrading typescript, and then work on this before 5.4 ships (~3 months)